druid/src/text/editable_text.rs

```diff
@@ -58,6 +58,12 @@ pub trait EditableText: Sized {
     /// Get the next codepoint offset from the given offset, if it exists.
     fn next_codepoint_offset(&self, offset: usize) -> Option<usize>;
 
+    /// Get the line beginning offset from the given offset
+    fn left_end_of_line(&self, offset: usize) -> usize;
+
+    /// Get the line ending offset from the given offset
+    fn right_end_of_line(&self, offset: usize) -> usize;
```

Comment:
Ugh I'm sorry, I screwed up here, was thinking somehow of an API a level higher in the stack.

These should really be `preceding_line_break` and `next_line_break`. Whether or not they are on the left or the right will depend on  the writing direction.